lib/compare_with_wikidata.rb

```diff
@@ -107,8 +108,14 @@ def expanded_wikitext(page_title)
       result.data['wikitext']
     end
 
+    def fix_encoding(s)
+      return s unless s.encoding == Encoding::ASCII_8BIT
+      s.dup.force_encoding(s.detect_encoding[:ruby_encoding])
+    end
+
```

Comment:
I'm not 100% convinced I'm not missing something here, but I think you can replace this with charlock holmes' `.detect_encoding!` method. 

If you're avoiding that because that's only available in ruby 1.9+, I think it's fine to make that a pre-requisite here too. If there's something subtler going on where this isn't actually equivalent, then we should have a test to catch that case, as they currently pass with that change.